Use persistent-claim ReadOnly flag when mounting PVs

markturansky opened the pull request:
Resolves #10244 

A valid use case is GCEPD:   Mounted once read/write or mounted many times read-only.

@thockin 

will add unit tests for each volume effected.

markturansky committed:
Honor ReadOnly flag from persistent-volume plugin

markturansky committed:
IsReadOnly bool on builder

zmerlynn commented:
WIP still?

zmerlynn commented:
@k8s-bot retest this please

zmerlynn commented:
Assigning to @thockin to sequester or review. (Or close if it stays WIP for too much longer.)

thockin commented:
This is pretty on-the-edge.  I'm inclined to push it to after 1.0, but I'd like a second opinion.  @quinton-hoole - without this, a GCEPD can not be mounted read-only for multiple readers.  I think it can wait until post 1.0

zmerlynn commented:
Customer hat: I could wait for that fix. It takes a kind of advanced shared-PD use case.

markturansky committed:
Added unit tests for each PV using IsReadOnly

markturansky commented:
WIP until unit tests, which I just added.  I tend to agree this is an advanced use case.

The volumes have private implementations with private readOnly booleans.   I added an IsReadOnly func to Builder to provide access to that flag.  

Feedback is welcome :)

markturansky committed:
rebased and updated to latest

markturansky committed:
rebased and updated

thockin commented:
LGTM

@saad-ali since we were talking about read-only today

markturansky commented:
@saad-ali interested in a review so I can get a LGTM?

saad-ali commented:
I'll take a look

saad-ali commented:
@markturansky could you rebase please

markturansky committed:
rebased and updated

markturansky commented:
@saad-ali rebased

saad-ali commented:
LGTM

brendandburns commented:
@k8s-bot test this [testing build queue, sorry for the noise]

brendandburns commented:
Automatic merge from SubmitQueue

derekwaynecarr commented:
We need a consistent squash commit policy moving forward.

Having PRs that merge commits where 3 of the 6 are just "rebased and updated" makes our logs really really really hard to follow.

Can we get a formal policy queued up on this?

markturansky commented:
@derekwaynecarr agreed.  sorry about that.

pmorie commented:
+1 to squash commit policy; let's discuss this on the next community hangout.

thockin commented:
We've actually discussed this internally, it's just hard to enforce and
easy to forget.  The policy goes something like this:

Upon merge, all commits should be meaningful units of work.  Use commits to
add clarity to the development and review process.  Before merging, squash
any "fix review feedback", "typo", and "rebased" sorts of commits.  It is
not imperative that every commit in a PR compile and pass tests
independently, but it is worth striving for.  For mass automated fixups
(e.g. automated doc formatting), use one or more commits for the changes to
tooling and a final commit to apply the fixup en masse.  This makes reviews
much easier.

Is it worth writing that down?  As with all rules - enforcement is the key.



On Thu, Jul 30, 2015 at 2:05 PM, Paul Morie <notifications@github.com>
wrote:

> +1 to squash commit policy; let's discuss this on the next community
> hangout.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/10424#issuecomment-126487167>
> .
>
